gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/SpecimenMapper.java

```diff
@@ -0,0 +1,134 @@
+        getType(specimenCompoundStatement).ifPresent(specimen::setType);
+        getCollectedDateTime(specimenCompoundStatement).ifPresent(specimen::setCollection);
+
+        return specimen;
+    }
+
+    private Optional<SpecimenCollectionComponent> getCollectedDateTime(RCMRMT030101UK04CompoundStatement specimenCompoundStatement) {
+        var specimenRoleOpt = getSpecimenRole(specimenCompoundStatement);
+        if (specimenRoleOpt.isPresent() && specimenRoleOpt.get().getEffectiveTime() != null) {
+            var effectiveTime = specimenRoleOpt.get().getEffectiveTime();
+            if (effectiveTime.hasCenter()) {
+                var effective = dateTimeMapper.mapDateTime(effectiveTime.getCenter().getValue());
+                return Optional.of(new SpecimenCollectionComponent().setCollected(effective));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private List<Annotation> getNote(RCMRMT030101UK04CompoundStatement specimenCompoundStatement) {
+        return specimenCompoundStatement.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component02::getNarrativeStatement)
+            .filter(Objects::nonNull)
+            .map(narrativeStatement -> getLastLine(narrativeStatement.getText()))
+            .map(e -> new Annotation().setText(e))
+            .toList();
+    }
+
+    private Optional<CodeableConcept> getType(RCMRMT030101UK04CompoundStatement specimenCompoundStatement) {
+        var specimenRoleOpt = getSpecimenRole(specimenCompoundStatement);
+        if (specimenRoleOpt.isPresent()) {
+            var specimenMaterialOpt = Optional.ofNullable(specimenRoleOpt.get().getSpecimenSpecimenMaterial());
+            if (specimenMaterialOpt.isPresent()) {
+                if (specimenMaterialOpt.get().getDesc() != null) {
+                    return Optional.of(new CodeableConcept().setText(specimenMaterialOpt.get().getDesc()));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<Identifier> getAccessionIdentifier(RCMRMT030101UK04CompoundStatement specimenCompoundStatement) {
+        var specimenRoleOpt = getSpecimenRole(specimenCompoundStatement);
+        if (specimenRoleOpt.isPresent()) {
+            var specimenIds = specimenRoleOpt.get().getId();
+            if (specimenIds.size() > 1) {
+                return Optional.of(new Identifier().setValue(specimenIds.get(1).getExtension()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<RCMRMT030101UK04SpecimenRole> getSpecimenRole(RCMRMT030101UK04CompoundStatement specimenCompoundStatement) {
+        return !specimenCompoundStatement.getSpecimen().isEmpty()
+            ? Optional.ofNullable(specimenCompoundStatement.getSpecimen().get(0).getSpecimenRole())
+            : Optional.empty();
+    }
+
+    private Optional<RCMRMT030101UK04CompoundStatement> getParentCompoundStatementByChildId(
```

Comment:
be careful with this, if your compound statement is nested it can throw an NPE here, which should be resolvable if you use the extractAllCompoundStatements method again.